lib/auth.js

```diff
@@ -98,7 +98,79 @@ function * loginSSO ({save}) {
   return {token: token, email: account.email}
 }
 
+function * logout () {
+  let token = cli.heroku.options.token
+  if (token) {
+    // for SSO logins we delete the session since those do not show up in
+    // authorizations because they are created a trusted client
+    let sessionsP = cli.heroku.delete('/oauth/sessions/~')
+    .catch(err => {
+      if (err.statusCode === 404 && err.body && err.body.id === 'not_found' && err.body.resource === 'session') {
+        return null
+      }
+      if (err.statusCode === 401 && err.body && err.body.id === 'unauthorized') {
+        return null
+      }
+      throw err
+    })
+
+    // grab the default authorization because that is the token shown in the
+    // dashboard as API Key and they may be using it for something else and we
+    // would unwittingly break an integration that they are depending on
+    let defaultAuthorizationP = cli.heroku.get('/oauth/authorizations/~')
+    .catch(err => {
+      if (err.statusCode === 404 && err.body && err.body.id === 'not_found' && err.body.resource === 'authorization') {
+        return null
+      }
+      if (err.statusCode === 401 && err.body && err.body.id === 'unauthorized') {
+        return null
+      }
+      throw err
+    })
+
+    // grab all the authorizations so that we can delete the token they are
+    // using in the CLI.  we have to do this rather than delete ~ because
+    // the ~ is the API Key, not the authorization that is currently requesting
+    let authorizationsP = cli.heroku.get('/oauth/authorizations')
+    .catch(err => {
+      if (err.statusCode === 401 && err.body && err.body.id === 'unauthorized') {
+        return []
+      }
+      throw err
+    })
+
+    /* eslint-disable no-unused-vars */
+    let [_, defaultAuthorization, authorizations] = yield [sessionsP, defaultAuthorizationP, authorizationsP]
```

Comment:
you can avoid this by doing `let [, defaultAuthorization, authorizations] = yield [sessionsP, defaultAuthorizationP, authorizationsP]` I think